packages/lesswrong/server/codegen/typeGenerationUtils.ts

```diff
@@ -20,16 +20,26 @@ function maybeNullable(type: string, nullable: boolean) {
   return nullable ? `${type} | null` : type 
 }
 
-export function simplSchemaTypeToTypescript(schema, fieldName, simplSchemaType): string {
+export function simplSchemaTypeToTypescript(schema: SchemaType<DbObject>, fieldName: string, simplSchemaType, indent = 2): string {
   if (simplSchemaType.singleType == Array) {
     const elementFieldName = `${fieldName}.$`;
-    if (!(elementFieldName in schema))
+    if (!(elementFieldName in schema)) {
       throw new Error(`Field ${fieldName} has an array type but ${fieldName}.$ is not in the schema`);
+    }
+
     const typescriptStrElementType = simplSchemaTypeToTypescript(schema, elementFieldName, schema[elementFieldName].type);
     return `Array<${typescriptStrElementType}>`;
   } else if (simplSchemaType.singleType) {
-    const nullable = !!schema[fieldName]?.nullable
-    if (simplSchemaType.singleType == String) return maybeNullable("string", nullable);
+    const allowedValues = simplSchemaType.definitions[0]?.allowedValues;
+    const nullable = !!schema[fieldName]?.nullable;
+
+    if (simplSchemaType.singleType == String) {
```

Comment:
We only have `allowedValues` being used for string literals, but I'm open to having it cover numeric literals too.